### DISCUSSION TITLE
Error deleting item from fieldArray after validating with YUP

### DISCUSSION THREAD

**Discussion by Ivandeclei** [react-hook-form/react-hook-form#4076] [Q&A]
Hello I'm using useFieldArray, we use the remove method from useFieldArray.

the function is as follows.

`const remover = async (index){
        await fieldArrayMethods.remove(index);
	setShowConfirmarExclusao(null)
        
    }
`

while there is no validation with YUP everything works fine, there is a normal exclusion of each item, but when there is a validation and there is a validation error by YUP and then the attempt to delete presents the following error.

Unhandled Rejection (TypeError): data.slice is not a function or its return value is not iterable.

can you help me?

**Comment by bluebill1049** [comment]
can you share a codesandbox?

**Reply by Ivandeclei** [reply]
sorry, but i can't share the code due to restrictions where i work.

but I can share some prints of the code and the error.

can it be like this ?

**Reply by bluebill1049** [reply]
which version are you using?

**Reply by Ivandeclei** [reply]
we are using version 5

below is a gif with the error happening.

![error](https://user-images.githubusercontent.com/33641590/106750879-3088ad80-6607-11eb-9ad4-6bb503884f65.gif)

below where each input is created

select

![02 - select](https://user-images.githubusercontent.com/33641590/106751400-e227de80-6607-11eb-8768-ab1f36cca67d.PNG)

text 
![01 - inputgenerico](https://user-images.githubusercontent.com/33641590/106752822-a3932380-6609-11eb-9a31-dd067476674e.PNG)

below where each line of the fieldArray is generated
![01 - topoFieldArray](https://user-images.githubusercontent.com/33641590/106752859-adb52200-6609-11eb-80fb-45a6f6effbcf.PNG)

add and remove functions
![02 - adicionar e remover fieldArray](https://user-images.githubusercontent.com/33641590/106752933-c32a4c00-6609-11eb-9025-04b44b46fb95.PNG)

useEffect
![03 - useeffect](https://user-images.githubusercontent.com/33641590/106752975-d0473b00-6609-11eb-8e3e-d93a71f84c89.PNG)

below where the fieldarray component is mounted
![04 - criacao componente](https://user-images.githubusercontent.com/33641590/106753078-efde6380-6609-11eb-9b2a-e35f083cca3c.PNG)

below the error that occurs when yup validation happens and there is an attempt to delete a line from the fieldArray
![erro apresentado](https://user-images.githubusercontent.com/33641590/106753194-1ef4d500-660a-11eb-8697-cfa7f38e78e0.PNG)

**Reply by bluebill1049** [reply]
i think there is problem with v5 aysnc, remove that await see if the issue move away?

**Reply by Ivandeclei** [reply]
the error remains even removing async and await

![image](https://user-images.githubusercontent.com/33641590/106889630-0bf40a80-66c7-11eb-8918-4b39adf33f10.png)

![image](https://user-images.githubusercontent.com/33641590/106889700-262de880-66c7-11eb-8ed9-831fbe8a7b13.png)

**Reply by bluebill1049** [reply]
common out `setXXXX` still got issue? if so provide a codesandbox.
